contrib/babelfishpg_tsql/sql/sys_function_helpers.sql

```diff
@@ -9727,28 +9727,37 @@ $BODY$
 LANGUAGE plpgsql
 STABLE;
 
-CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(IN arg anyelement,
+CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(IN typmod INTEGER,
+                                                                  IN arg anyelement,
                                                                   IN try BOOL,
                                                                   IN p_style NUMERIC DEFAULT 0)
 RETURNS sys.varbinary
 AS
 $BODY$
+DECLARE result sys.varbinary;
 BEGIN
     IF try THEN
-        RETURN sys.babelfish_try_conv_to_varbinary(arg, p_style);
+        RETURN sys.babelfish_try_conv_to_varbinary(typmod, arg, p_style);
     ELSE
         IF pg_typeof(arg) IN ('text'::regtype, 'sys.ntext'::regtype, 'sys.nvarchar'::regtype, 'sys.bpchar'::regtype, 'sys.nchar'::regtype) THEN
             RETURN sys.babelfish_conv_string_to_varbinary(arg, p_style);
```

Comment:
What about these types? Do we return correct typmod for these inputs?

Reply:
This is being verified , by the authors adding comprehensive tests for varbinary/binary datatypes.